### PR TITLE
CORCI-445 Adding node provisioning

### PR DESCRIPTION
**Pull request by JohnMalmberg**
* Add provisionNodes step.
  Allows snapshot option for VMs, and a power cycle only option.
  By default waits for the provisioning or power cycle to complete.

* Update checkPatch step credential ID.

Signed-off-by: John.E.Malmberg <john.e.malmberg@intel.com>

### CODE REVIEW

File: vars/checkPatch.groovy [hunk 1]
```diff
@@ -46,7 +46,7 @@ def call(Map config = [:]) {
     // Need the jenkins module to do linting
     checkoutScm url: 'ssh://review.hpdd.intel.com:29418/exascale/jenkins',
                 checkoutDir: 'jenkins',
-                credentialsId: 'bf21c68b-9107-4a38-8077-e929e644996a'
+                credentialsId: 'daos-gerrit-read'
```
Comment (brianjmurrell) [inline vars/checkPatch.groovy]: These have the equivalent credentials in jenkins-3 then?

Comment (JohnMalmberg) [reply vars/checkPatch.groovy]: Yes, the credential ID has been installed on Jenkins-3.

File: vars/provisionNodes.groovy [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                              cat /tmp/id_rsa.pub >> /localhome/jenkins/.ssh/authorized_keys
+                              mv /tmp/id_rsa /localhome/jenkins/.ssh/
+                              chmod 700 /localhome/jenkins/.ssh
+                              chmod 600 /localhome/jenkins/.ssh/{authorized_keys,id_rsa}
+                              chown -R jenkins.jenkins /localhome/jenkins/.ssh
+                              echo \\"jenkins ALL=(ALL) NOPASSWD: ALL\\" > /etc/sudoers.d/jenkins
+                              yum copr -y enable jhli/ipmctl
+                              yum copr -y enable jhli/safeclib
+                              yum install -y libipmctl-devel openmpi CUnit fuse python34-PyYAML python34-nose python34-pip valgrind python34-paramiko'''
+    sh script: 'set -x; pdcp -R ssh -l root -w ' + config['NODELIST'] +
```
Comment (JohnMalmberg) [inline vars/provisionNodes.groovy]: This needs to be reworked.
SSH private keys access needs to be from ssh-agent(credentials: credential_id) only.
The daos-provisioner ID can be used for privileged access to the CI nodes.

Comment (brianjmurrell) [reply vars/provisionNodes.groovy]: I changed that to [generate a session key](https://github.com/daos-stack/pipeline-lib/pull/13/commits/313151f2769576236c676cc0407956eb4db9f4f7).

**Pull request merged**